src/AndcultureCode.CSharp.Sitefinity.Core/JsonSerialization/Converters/SitefinityEnumConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AndcultureCode.CSharp.Sitefinity.Core.JsonSerialization.Converters
+{
+    /// <summary>
+    /// Responsible for generically converting enums into values that correspond with how Sitefinity handles
+    /// ChoiceField indices. See https://www.progress.com/documentation/sitefinity-cms/choicefield-operations
+    /// for more information on how the ChoiceField works through the OData web services.
+    /// 
+    /// Developer Note: This converter operates under the assumption that the specified Enum has values starting
+    /// at 1 NOT 0.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class SitefinityEnumConverter<T> : JsonConverter where T : struct, IConvertible
```

Comment:
I know this was something I originally added, but I'm wondering if this should be renamed to `SitefinityChoiceFieldConverter` or something similar since the logic here is based around Sitefinity's `ChoiceField`